heat/core/communication.py

```diff
@@ -103,31 +119,420 @@ def chunk(self, shape, split):
             tuple(shape[i] if i != split else end - start for i in range(dims)), \
             tuple(slice(0, shape[i]) if i != split else slice(start, end) for i in range(dims))
 
-    @staticmethod
-    def as_buffer(obj):
-        if isinstance(obj, tensor.tensor):
-            obj = obj._tensor__array
-        if not isinstance(obj, torch.Tensor):
-            return obj
+    def counts_displs_shape(self, shape, axis):
+        """
+        Calculates the item counts, displacements and output shape for a variable sized all-to-all MPI-call (e.g.
+        MPI_Alltoallv). The passed shape is regularly chunk along the given axis and for all nodes.
+
+        Parameters
+        ----------
+        shape : tuple(int)
+            The object for which to calculate the chunking.
+        axis : int
+            The axis along which the chunking is performed.
+
+        Returns
+        -------
+        counts_and_displs : two-tuple of tuple of ints
+            The counts and displacements for all nodes
+        """
+        # the elements send/received by all nodes
+        counts = np.full((self.size,), shape[axis] // self.size)
+        counts[:shape[axis] % self.size] += 1
+
+        # the displacements into the buffer
+        displs = np.zeros((self.size,), dtype=counts.dtype)
+        np.cumsum(counts[:-1], out=displs[1:])
+
+        # helper that calculates the output shape for a receiving buffer under the assumption all nodes have an equally
+        # sized input compared to this node
+        output_shape = [ele for ele in shape]
```

Comment:
list(shape)

Reply:
Curious I cannot reproduce this, for how many cores? mpirun -np 1 python -m unittest

Reply:
I'm late to this so maybe you have solved it already, but we have modified __eq__ to return a ht.bool instead of a torch.uint8, that's why the argmin test fails (#146)

Reply:
That might be it. I will have a look into it, thanks

Reply:
i think it's a problem with the logic of the test and not the argmin or all functions: a print shows that the comparison is between 'tensor([[1, 1, 1, 1, 1, 1, 1, 1]])' and  '8' with -np 4.

Reply:
Issue has been resolved by a fix, thanks for the remark!